sky/packages/sky/lib/widgets/popup_menu.dart

```diff
@@ -112,63 +91,71 @@ class PopupMenu extends AnimatedComponent {
       boxShadow: shadows[level]));
   }
 
-  void _onStatusChanged(AnimationStatus status) {
-    if (status == AnimationStatus.dismissed &&
-        navigator != null &&
+  void _onDismissed() {
+    if (navigator != null &&
         navigator.currentRoute is RouteState &&
         (navigator.currentRoute as RouteState).owner == this) // TODO(ianh): remove cast once analyzer is cleverer
       navigator.pop();
     if (onStatusChanged != null)
-      onStatusChanged(status);
-  }
-
-
-  void _open() {
-    _performance.play();
-    if (navigator != null)
-      navigator.pushState(this, (_) => _close());
-  }
-
-  void _close() {
-    _performance.reverse();
+      onStatusChanged(AnimationStatus.dismissed);
   }
 
   BoxPainter _painter;
 
   Widget build() {
-    int i = 0;
-    List<Widget> children = new List.from(items.map((Widget item) {
-      return new Opacity(opacity: _itemOpacities[i++].value, child: item);
-    }));
+    double unit = 1.0 / (items.length + 1.5); // 1.0 for the width and 0.5 for the last item's fade.
+    List<Widget> children = [];
+    for (int i = 0; i < items.length; ++i) {
+      double start = (i + 1) * unit;
+      double end = (start + 1.5 * unit).clamp(0.0, 1.0);
+      children.add(new FadeTransition(
+        direction: showing ? Direction.forward : Direction.reverse,
+        performance: _performance,
+        opacity: new AnimatedValue<double>(0.0, end: 1.0, interval: new Interval(start, end)),
```

Comment:
build is no longer called every frame. Only when the PopupMenu itself changes. The transition classes handle the per-frame building, now. (One of the goals of this patch.)